Loop/View Controllers/StatusTableViewController.swift

```diff
@@ -554,9 +554,20 @@ final class StatusTableViewController: LoopChartsTableViewController {
                 self.statusCharts.preMealOverride = self.deviceManager.loopManager.settings.preMealOverride
                 self.statusCharts.scheduleOverride = self.deviceManager.loopManager.settings.scheduleOverride
             }
-            if self.statusCharts.scheduleOverride?.hasFinished() == true {
+            if let scheduleOverride = self.statusCharts.scheduleOverride,
+               scheduleOverride.hasFinished() == true
+            {
+                self.deviceManager.loopManager.mutateSettings { settings in
+                    settings.clearOverride(matching: scheduleOverride.context)
```

Comment:
I don't think this is an issue where the *state* needs to be updated (the state already indicates the end time in this situation), but rather it's a display issue. The status table view controller can set a timer to trigger on the end time of the override, and refresh then (state still shouldn't need to be updated, as any users of this attribute should be paying attention to end time).

Reply:
Just so I understand, the override in settings should remain even though it has finished?

Reply:
Yes, I believe that is fine.